### PR TITLE
adjust to 2.13.6 restriction on narrowing access when overriding

### PR DESCRIPTION
**Pull request by SethTisue**
as per scala/scala#9542

this came up in the Scala 2.13 community build

### CODE REVIEW

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=develop&zinc=develop&io=pull/317/head&librarymanagement=develop&util=develop&website=develop&deploy_to_bintray=false&run_sbt_community=false&verbose_dbuild=false&community_verbose=false">this link.</A>

**Comment by SethTisue** [comment]
I initially got a spurious failure on MacOS that went away when I re-ran the jobs.

I added the `fail-fast` thing in a second commit to prevent such a spurious failure in one job from causing the other jobs to be canceled.

File: io/src/main/scala/sbt/nio/file/Glob.scala [hunk 1]
```diff
@@ -903,11 +903,11 @@ object RelativeGlob {
     override def hashCode: Int = glob.hashCode
   }
   private final class GlobMatcher(override val glob: String) extends SingleComponentMatcher {
-    private[this] val (prefix, pattern) = glob.indexOf(":") match {
+    private[this] val (prefixString, pattern) = glob.indexOf(":") match {
```
Comment (som-snytt) [inline io/src/main/scala/sbt/nio/file/Glob.scala:906]: Could also push down to initializer of `matcher`.

**Review by eed3si9n — APPROVED** [review]
LGTM

**Pull request merged**